src/Plugin/Patches.php

```diff
@@ -146,12 +123,44 @@ public function activate(Composer $composer, IOInterface $io): void
                 'type' => 'string',
                 'default' => 'patches.json',
             ],
+            'patches-lock-file' => [
+                'type' => 'string',
+                'default' => 'patches.lock.json',
+            ],
             "ignore-dependency-patches" => [
                 'type' => 'list',
                 'default' => [],
             ],
         ];
         $this->configure($this->composer->getPackage()->getExtra(), 'composer-patches');
+        $this->lockFile = new JsonFile(
+            $this->getPatchesLockFilePath(),
+            null,
+            $this->io
+        );
+        $this->locker = new Locker($this->lockFile);
+    }
+
+    /**
+     * Get the path to the current patches lock file.
+     */
+    public function getPatchesLockFilePath(): string
+    {
+        $composer_file = \Composer\Factory::getComposerFile();
+
+        $dir = dirname(realpath($composer_file));
+        $base = pathinfo($composer_file, \PATHINFO_FILENAME);
+
+        $lock_file = $this->getConfig('patches-lock-file');
+        if (!is_string($lock_file)) {
+            $lock_file = 'patches-lock-file';
+        }
+
+        if ($base === 'composer') {
+            return "$dir/$lock_file";
+        }
+
+        return "$dir/$base-$lock_file";
```

Comment:
Right now, if you set `COMPOSER=composer-foo.json` and run any of the composer patches commands, you'll get `composer-foo-patches.lock.json`. If we set `patches-lock-file` to `asdf.lock.json` and _also_ set `COMPOSER=composer-foo.json`, I think we'll get `composer-foo-asdf.lock.json`.

This isn't incorrect, but it _is_ something that docs and tests will need to account for. Just flagging for visibility.